### DISCUSSION TITLE
proposition for Key Exchange API functions name change.

### DISCUSSION THREAD

**Discussion by sh-dv** [jedisct1/libsodium#1127] [General]
I know that the [Key Exchange API](https://libsodium.gitbook.io/doc/key_exchange#example-client-side) is mostly suitable for a client/server scenario. But it basically can be used in any other way. Thus, functions named like `crypto_kx_client_session_keys` or `crypto_kx_server_session_keys` don't make sense when you use them for example to compute a shared key between two users for a chat app. Then the code won't make sense in such scheme because there is no server/client interaction, the key exchange API was used to compute a key for 2 parties.

What do you think? and do you suggest another naming?

**Comment by jedisct1** [comment]
The functions will not be renamed. That would break all existing applications with no practical benefits.

A shared key is computed between two parties, and this is opaque to what they actually represent in applications.

Calling them client and server matches the most common use case. But if they are called "A" and "B" in your application, feel free to map "A" to the "server" key and "B" to the "client" key or the other way round :)

What you get out of the function is two keys, that can be used in any context where having two keys make sense. You can also use only one of them if you only communicate in a single direction.

**Comment by sh-dv** [comment]
Totally understood, Thanks.
